sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -226,6 +226,7 @@ class SessionCatalog(
           s"${globalTempViewManager.database}.viewName.")
     }
     requireDbExists(dbName)
+    externalCatalog.setCurrentDatabase(dbName)
```

Comment:
> we always qualify the table names with the database names

It explains why we don't need this fix. IMO if it only impacts tests, we should fix it in tests.